core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestBeginTableWrite.java

```diff
@@ -132,14 +133,44 @@ public void testUpdateWithInvalidNode()
     private void applyOptimization(Function<PlanBuilder, PlanNode> planProvider)
     {
         Metadata metadata = new MockMetadata();
-        new BeginTableWrite(metadata, createTestingFunctionManager())
-                .optimize(
-                        planProvider.apply(new PlanBuilder(new PlanNodeIdAllocator(), metadata, testSessionBuilder().build())),
-                        testSessionBuilder().build(),
-                        empty(),
-                        new SymbolAllocator(),
-                        new PlanNodeIdAllocator(),
-                        WarningCollector.NOOP);
+        Session session = testSessionBuilder().build();
+        SymbolAllocator symbolAllocator = new SymbolAllocator();
+        PlanNodeIdAllocator idAllocator = new PlanNodeIdAllocator();
+
+        new BeginTableWrite(metadata, createTestingFunctionManager()).optimize(
+                planProvider.apply(new PlanBuilder(new PlanNodeIdAllocator(), metadata, testSessionBuilder().build())),
+                new PlanOptimizer.Context()
+                {
+                    @Override
+                    public Session getSession()
+                    {
+                        return session;
+                    }
+
+                    @Override
+                    public TypeProvider getTypeProvider()
+                    {
+                        return empty();
```

Comment:
that's incorrect.

why not `symbolAllocator.getTypes()`? 

let's remove `Context.getTypeProvider`